python/paddle/v2/fluid/layers.py

```diff
@@ -60,8 +60,13 @@ def fc(input,
         param_shape = [
             reduce(lambda a, b: a * b, input_shape[num_flatten_dims:], 1)
         ] + [size]
+
+        std = (2.0 / (param_shape[0]**2 * size))**0.5
         w = helper.create_parameter(
-            attr=param_attr, shape=param_shape, dtype=dtype)
+            attr=param_attr,
+            shape=param_shape,
+            dtype=dtype,
+            initializer=NormalInitializer(0.0, std, 1))
```

Comment:
How much is the effect of this initialization?
I know this change make it same as v2 initialization. But perhaps the default XavierInitializer is more common choice. We should allow the user to supply initializer. In fact, it is can be supplied in param_attr. Adding an initializer here will ignore the initializer setting in param_attr and cause confusing.

In order to make the sample mnist code, it is better to change the example to set the initializer through param_attr. And if the different std does not have much effect, we don't even need to bother to change the mnist example.

Reply:
Sorry for such a late reply caused by the trip. I compared the forward cost, accuracy, loss with v2/tensorflow batch by batch. Because our default Initializer is UniformInitializer, has a 
-0.02 compared with the benchmark at the first several batches.  With the NormalInitalizer w e have reached the same result with Paddle V2, even at the last decimal. I think we can draw the conclusion that our implementation has got the same accuracy with the Paddle V2.

Tensorflow uses a special random number generator algorithm `Philox` which we don't have. We can reach the approximal same result when we fill Variable with the same random value, but there is a difference at 6 to 8 decimal place, caused by the conv2d operator implementation difference. 

>We should allow the user to supply initializer. In fact, it is can be supplied in param_attr. Adding an initializer here will ignore the initializer setting in param_attr and cause confusing.

We are working on make Initializer configurable. . Sure will only change the mnist example configuration in the benchmark test and set fc layer with XavierInitilizer as default.
https://github.com/PaddlePaddle/Paddle/issues/5819
https://github.com/PaddlePaddle/Paddle/pull/5805
https://github.com/PaddlePaddle/Paddle/pull/5760